ow_plexil/src/plans/CheckForFaults.plp

```diff
@@ -0,0 +1,31 @@
+// The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+// Check for a variety of faults at regular intervals, and set a general health
+// variable accordingly.
+
+// Note that the fault queries (lookups) are for categories of faults, but that
+// the exact faults (outlined in ../plexil-adapter/OwInterface.h) are not yet
+// detectable in PLEXIL.
+
+#include "plan-interface.h"
+
+CheckForFaults:
+{
+  InOut Boolean all_ok;
+  End all_ok;
+
+  Iterate:
+  {
+    Repeat true;
+    all_ok = !(Lookup(SystemFault) ||
+               Lookup(AntennaFault) ||
+               Lookup(ArmFault) ||
+               Lookup(PowerFault));
+
+    log_info ("Waiting until there are no faults...");
+
+    Wait 5; // seconds, arbitrary choice
+  }
+}
```

Comment:
I'm afraid this is a pattern we should avoid in any PLEXIL plan: polling the environment.  PLEXIL plans should ideally be event-driven.   If the application interface (C++) is properly implemented, the Plexil exec is notified ("woken up") whenever there's a change to a Lookup of interest.   I recently fixed the missing notification for updates to SystemFault, AntennaFault, ArmFault, PowerFault, so the polling that existed in RM2 is no longer needed.

This plan is basically not needed.

A smaller point: SystemFault is a bitmask that subsumes all the other faults.  If you're checking this, the other checks are redundant.

Reply:
This plan is almost exactly like MonitorFaults.plp, which was used in RM2 and hence I didn't write, which is why I figured the redundancy was okay. The only difference is fewer inputs and the End condition.  I actually asked about the SystemFault redundancy a while back, so it good to confirm that now.

That said, I can see how just using the SystemFault lookup directly in the initialization plan is better since it is inherently all-encompassing.

Reply:
RM2 should be heavily revised after you've completed a few patterns -- it had some poor techniques in it, and was written before last week's critical bug fix.

Here's an idea: with each pattern that you complete, see how it can be employed to improve RM2.